February 2024 Release Notes (#1397)
semgrep/semgrep-docs: docs/release-notes/february-2024.md

    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: February 2024
tags:
  - Release notes
---

# Semgrep release notes for February 2024

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in February 2024:
  * [<i class="fas fa-external-link fa-xs"></i>1.59.1](https://github.com/semgrep/semgrep/releases/tag/v1.59.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.60.0](https://github.com/semgrep/semgrep/releases/tag/v1.60.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.60.1](https://github.com/semgrep/semgrep/releases/tag/v1.60.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.61.1](https://github.com/semgrep/semgrep/releases/tag/v1.61.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.62.0](https://github.com/semgrep/semgrep/releases/tag/v1.62.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.63.0](https://github.com/semgrep/semgrep/releases/tag/v1.63.0)

## 🌐 Cloud Platform

### Added

### Changed

### Fixed

## 💻 Code

### Added

* Added new rules for Elixir and the Phoenix framework, covering various security and correctness issues. These are available in the `p/elixir`

>   ruleset. <!-- Do we need to mention that this is for Pro users only? -->

Reviewer: Are these Semgrep Pro rules?

Author: Yup, unless I misunderstand this?

> To use them, users must be logged in and use the Pro engine via the --pro option! Take that, Semgrep OSS!

[Slack](https://semgrepinc.slack.com/archives/C03A5BETNS3/p1706885627454809)

Reviewer: Since they are Pro rules, I think we can just say "Added new Pro rules..." I'll make the suggestion.

  * [<i class="fas fa-external-link fa-xs"></i>1.59.1](https://github.com/semgrep/semgrep/releases/tag/v1.59.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.60.0](https://github.com/semgrep/semgrep/releases/tag/v1.60.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.60.1](https://github.com/semgrep/semgrep/releases/tag/v1.60.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.61.1](https://github.com/semgrep/semgrep/releases/tag/v1.61.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.62.0](https://github.com/semgrep/semgrep/releases/tag/v1.62.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.63.0](https://github.com/semgrep/semgrep/releases/tag/v1.63.0)

## 🌐 Cloud Platform

### Added

- API: Added a `rule` object under `findings` with the following fields: 
    - `name`
    - `message`
    - `confidence`
    - `category`
    - `subcategories`
    - `technologies`
    - `vulnerability_classes`
    - `cwe_names`
    - `owasp_names` <!-- 12868 --> 

### Changed

### Fixed

- Fixed a bug where the navigation sidebar covered mobile screens and could not be collapsed. <!-- 12876 -->

## 💻 Code

### Added

> * Added new rules for Elixir and the Phoenix framework, covering various security and correctness issues. These are available in the `p/elixir`

Reviewer: ```suggestion
* Added new Pro rules for Elixir and the Phoenix framework, covering various security and correctness issues. These are available in the `p/elixir`
```

  ruleset. <!-- Do we need to mention that this is for Pro users only? -->
* Added support for Python, with a focus on the Flask ecosystem, to the Semgrep
  Pro Engine.

> * Added ability to collapse and turn off patterns in a rule.

Reviewer: Is this structure mode? Did we release it?

Author: Yep, this is a part of structure mode; I forgot to delete this line.

> * Added the ability to distinguish between which rules are available to Semgrep

Reviewer: Do end users see this distinction somewhere?

Author: Made it clearer this is strictly referring to the Playground

Author: Yeah, it's this one: https://github.com/semgrep/semgrep-app/pull/12275

I'll remove this one in favor of your additions?

  Pro Engine users and which ones are available to Semgrep OSS users.

> * Added support for nested record patterns on the left-hand side of an

Reviewer: is this a language feature or a playground/editor mode feature?

Author: Dataflow analysis; I added an example

  assignment.
* `metavariable-regex` can now match on metavariables of interpolated strings
  that use variables with known values.

> * Added support for parsing Swift Package Manager manifest files and lock files.

Reviewer: I believe this is for SSC, could you please move it to that section?

* **Taint analysis**:
  * Added support for Python constructors

>   * Added support for index sensitivity. Semgrep tracks taint on individual

Reviewer: Half the list has full stops, and the other half has none, may I suggest using full stops?

### Changed

* Improved error handling during inter-file analysis so Semgrep Code doesn't crash.
* **CLI**: If there are multiple errors resulting from the user running Pro
  rules without a paid subscription, the CLI groups all errors and reports a
  single warning.
* The project name for repositories scanned locally is `local_scan/<repo_name>`
  instead of `<repo_name>`.
* The **View Results** URL displayed for findings now includes the repository
  and branch names.

### Fixed

* Fixed issues with trailing newline parsing in `pyproject.toml` and
  `poetry.lock` files.
* Fixed an issue with incorrect autofix application where multiple fixes were
  applied to the same line.
* Fixed issue where tokens for type parameter brackets weren't stored correctly.
  They're now stored in the generic AST, allowing Semgrep to autofix
  these constructs correctly.
* Fixed an issue where Semgrep doesn't support multiple labels for taint
  traces. Now, Semgrep looks at the `requires` of the sink, and if it has the
  shape `A and ...`, it picks `A` as the preferred label and reports the
  trace.
* Fixed issue where taint signatures don't capture changes to parameter fields.

> * Scan summary links printed after users run `semgrep ci` now reflect a

Reviewer: This belongs in the Semgrep Cloud Platform section :) `semgrep ci` can run any number of products, so it's not Code's, really. It's `semgrep scan` that only performs SAST scans.

  * [<i class="fas fa-external-link fa-xs"></i>1.63.0](https://github.com/semgrep/semgrep/releases/tag/v1.63.0)

## 🌐 Cloud Platform

### Added

- API: Added a `rule` object under `findings` with the following fields: 
    - `name`
    - `message`
    - `confidence`
    - `category`
    - `subcategories`
    - `technologies`
    - `vulnerability_classes`
    - `cwe_names`
    - `owasp_names` <!-- 12868 --> 
- Added distinction between Pro engine and OSS findings in the Playground and Editor. <!-- 12275 -->
- Support `linux-arm64` platform for Semgrep Pro Engine downloads <!-- 12430 -->

### Changed

- Updated the Semgrep Cloud Platform login page. <!-- 12744 -->
- Updated the login process from the CLI to SCP. This change affects new users. <!-- 12531 -->
- Updated the Semgrep installation instructions for Docker. <!-- 12531 -->
- Improved performance of Semgrep Playground and Editor. <!-- 12461 -->

### Fixed

- Fixed a bug where the navigation sidebar covered mobile screens and could not be collapsed. <!-- 12876 -->

## 💻 Code

### Added

<!-- I think the highest item should probably be our C, C++ launch? -->

> * Added new rules for Elixir and the Phoenix framework, covering various security and correctness issues. These are available in the `p/elixir`

Reviewer: ```suggestion
* Added new Pro rules for Elixir and the Phoenix framework, covering various security and correctness issues. These are available in the `p/elixir`
```

  ruleset. <!-- Do we need to mention that this is for Pro users only? -->
* Added support for Python, with a focus on the Flask ecosystem, to the Semgrep
  Pro Engine.
* Added ability to collapse and turn off patterns in a rule.
* Added the ability to distinguish between which rules are available to Semgrep
  Pro Engine users and which ones are available to Semgrep OSS users.
* Added support for nested record patterns on the left-hand side of an
  assignment.
* `metavariable-regex` can now match on metavariables of interpolated strings
  that use variables with known values.
* Added support for parsing Swift Package Manager manifest files and lock files.
* **Taint analysis**:
  * Added support for Python constructors
  * Added support for index sensitivity. Semgrep tracks taint on individual
    indexes of a data structure when these are constant values, either integers
    or strings, and the code uses the built-in syntax for array indexing.

>   * Added `exact: false` so you can specify that anything inside a code

Reviewer: ```suggestion
  * Added `exact: false` so you can specify that anything inside a code
```
I believe this is nested under one of the [required fields](https://semgrep.dev/docs/writing-rules/rule-syntax/#required)? Do you know if there's an example rule using `exact: false` so we can check?

    region is a sink.
  * When `exact: true` and `taint_assume_safe_functions: true`, Semgrep now
    considers that, if the specified formula isn't a `patterns` with a
    `focus-metavariable`, it must look for taint in the function call's arguments.

### Changed

> * Improved error handling during inter-file analysis so Semgrep Code doesn't crash.

Reviewer: ```suggestion
* Improved error handling during interfile analysis so Semgrep Code doesn't crash.
```
Let's prefer interfile over inter-file. One of the many things a linter could catch for us.

* **CLI**: If there are multiple errors resulting from the user running Pro

>   rules without a paid subscription, the CLI groups all errors and reports a

Reviewer: ```suggestion
  rules without a license, the CLI groups all errors and reports a
```
Suggest license over paid subscription because "license" is used in our other sales/marketing collateral.

  single warning.
* The project name for repositories scanned locally is `local_scan/<repo_name>`
  instead of `<repo_name>`.
* The **View Results** URL displayed for findings now includes the repository
  and branch names.

### Fixed

> * Fixed issues with trailing newline parsing in `pyproject.toml` and

Reviewer: This is for SSC :)

  `poetry.lock` files.
* Fixed an issue with incorrect autofix application where multiple fixes were
  applied to the same line.
* Fixed issue where tokens for type parameter brackets weren't stored correctly.
  They're now stored in the generic AST, allowing Semgrep to autofix
  these constructs correctly.
* Fixed an issue where Semgrep doesn't support multiple labels for taint
  traces. Now, Semgrep looks at the `requires` of the sink, and if it has the
  shape `A and ...`, it picks `A` as the preferred label and reports the
  trace.
* Fixed issue where taint signatures don't capture changes to parameter fields.
* Scan summary links printed after users run `semgrep ci` now reflect a
  custom `SEMGREP_APP_URL` if set.

## ⛓️ Supply Chain

### Added

* Added the ability to filter for dependencies that Semgrep has commented on.
  <!-- https://github.com/semgrep/semgrep-app/pull/12898 -->
* Added manual review advice to GitHub PR comments. Certain Semgrep Supply Chain (SSC) findings require **manual review** to verify if the finding is reachable or not. GitHub PR comments now include this advice to help you ascertain if the finding is reachable or not. <!-- 12907 -->

> * Separated `reachable_if` and `upgrade_only` exposures. Users can better determine a finding's criticality through this distinction. <!-- 12657 -->

Reviewer: Sara is waiting for clarification if the separation appears to the end user in some way in the UI.

Reviewer: It is not visible in the UI. removing.

## 🤖 Assistant (beta)

### Added

> - Added weekly emails summarizing Semgrep tasks, highlighting top priority items. <!-- Private beta - not sure if we should include this --> 

Reviewer: I also have a message waiting for clarification on this.

Reviewer: Removed. Resolving.

---
slug: february-2024
hide_title: true
description: >-
    Release notes include the changes, fixes, and additions in specific versions of Semgrep.
toc_max_heading_level: 3
title: February 2024
tags:
  - Release notes
---

# Semgrep release notes for February 2024

## 🔧 OSS Engine

* The following versions of the OSS Engine were released in February 2024:
  * [<i class="fas fa-external-link fa-xs"></i>1.59.1](https://github.com/semgrep/semgrep/releases/tag/v1.59.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.60.0](https://github.com/semgrep/semgrep/releases/tag/v1.60.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.60.1](https://github.com/semgrep/semgrep/releases/tag/v1.60.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.61.1](https://github.com/semgrep/semgrep/releases/tag/v1.61.1)
  * [<i class="fas fa-external-link fa-xs"></i>1.62.0](https://github.com/semgrep/semgrep/releases/tag/v1.62.0)
  * [<i class="fas fa-external-link fa-xs"></i>1.63.0](https://github.com/semgrep/semgrep/releases/tag/v1.63.0)

## 🌐 Cloud Platform

### Added

> - API: Added a `rule` object under `findings` with the following fields: 

Reviewer 2: @s-santillan should we link out to the API docs here? https://semgrep.dev/api/v1/docs/#tag/Finding/operation/semgrep_app.core_exp.findings.handlers.issue.openapi_list_recent_issues

Reviewer: Added!

    - `name`
    - `message`
    - `confidence`
    - `category`
    - `subcategories`
    - `technologies`
    - `vulnerability_classes`
    - `cwe_names`
    - `owasp_names` <!-- 12868 --> 

> - Added distinction between Pro engine and OSS findings in the Playground and Editor. <!-- 12275 -->

Reviewer 2: @s-santillan @milanwilliams We should remove "Pro engine" and think of better copy to indicate it's Semgrep Commercial. I also wonder if we just skip this for the release notes.

Reviewer: I left a similar clarifying question in Slack but posting here too just in case: 

Are we still referring to our commercial offerings as "Semgrep Pro Engine" and "Pro rules" in the app, marketing page, and registry?

Reviewer: Resolved with:

> Let’s keep it the same for now and consistent with the app.

No need for any changes.

- Added support for the `linux-arm64` platform when you download Semgrep Pro Engine. <!-- 12430 -->

### Changed

- Updated the Semgrep Cloud Platform (SCP) login page. <!-- 12744 -->
- Updated the login process from the CLI to SCP. This change affects new users. <!-- 12531 -->
- Updated the Semgrep installation instructions for Docker. <!-- 12531 -->
- Improved performance of Semgrep Playground and Editor. <!-- 12461 -->

### Fixed

- Fixed a bug where the navigation sidebar covered the entire mobile screen and could not be collapsed. <!-- 12876 -->
- Scan summary links printed after users run `semgrep ci` now reflect a
  custom `SEMGREP_APP_URL` if set.

## 💻 Code

### Added

> * Support for C and C++ is now generally available (GA).

Reviewer 2: Could we specify interfile analysis, proprietary rules, etc? cc @milanwilliams

Reviewer: AFAIK we did not add proprietary rules, but I did add that we have both cross-file and cross-function analysis.

### Fixed

* Fixed an issue with incorrect autofix application where multiple fixes were
  applied to the same line.
* Fixed issue where tokens for type parameter brackets weren't stored correctly.
  They're now stored in the generic AST, allowing Semgrep to autofix
  these constructs correctly.
* Fixed an issue where Semgrep doesn't support multiple labels for taint
  traces. Now, Semgrep looks at the `requires` of the sink, and if it has the
  shape `A and ...`, it picks `A` as the preferred label and reports the
  trace.
* Fixed issue where taint signatures don't capture changes to parameter fields.

## ⛓️ Supply Chain

### Added

* Added support for parsing Swift Package Manager manifest files and lock files.
* Added the ability to filter for dependencies that Semgrep has commented on.
  <!-- https://github.com/semgrep/semgrep-app/pull/12898 -->
* Added manual review advice to GitHub PR comments. Certain Semgrep Supply Chain (SSC) findings require **manual review** to verify if the finding is reachable or not. <!-- 12907 -->

### Fixed

* Fixed issues with trailing newline parsing in `pyproject.toml` and
  `poetry.lock` files.

## 🤖 Assistant (beta)

### Added

> - Added weekly emails summarizing Semgrep tasks, highlighting top priority items. <!-- Private beta - not sure if we should include this --> 

Reviewer 2: Let's skip private betas

> ## 🔐 Secrets (beta)

Reviewer 2: Let's remove beta from here

* Fixed issues with trailing newline parsing in `pyproject.toml` and
  `poetry.lock` files.

## 🤖 Assistant (beta)

### Added

- Added weekly emails summarizing Semgrep tasks, highlighting top priority items. <!-- Private beta - not sure if we should include this --> 

## 🔐 Secrets (beta)

### Added

- Added the following new rules:
  - Detection rules for Azure and AWS
  - Semantic secrets rules for Python, JavaScript, and TypeScript
  - Semantic rules for hard-coded credentials in bash for `curl` commands
- Added non-validator regex detection for databases, including MongoDB,
  Microsoft SQL Server, MySQL, Postgres, and Redis
- Added secrets rule management, which is accessible in Semgrep Cloud Platform
  by going to **Rules** > **Policies** > **Secrets**. This allows you to:
  - See all available rules
  - Set valid finding modes for the rules
  - Set invalid and error validation state modes across multiple rules

### Fixed

- Fixed an issue where the **Analysis method** filter in Semgrep Cloud Platform
  wasn't filtering correctly.

## 📝 Documentation and knowledge base

### Added

- The Semgrep docs sidebar has been reorganized to help users browse through the docs.

> - A [series of guides](/deployment/core-deployment/) to setting up Semgrep as part of a security program for your organization.

Author: Maybe add "Added..." to the beginning of each entries? I think this verb could be omitted if we didn't have the first sentence mentioning the sidebar reorg, though

Reviewer: I will add "added"!

- A guide to setting up a [network broker](/semgrep-ci/network-broker/) that facilitates secure access between Semgrep and your private network.
- [Experimental rules](/writing-rules/experiments/pattern-syntax/) syntax reference.
- The following knowledgebase articles:
    - [GitLab "Job's log exceeded limit" error](/kb/semgrep-ci/collect-gitlab-logs)
    - [Set up Jenkins pipeline projects for Bitbucket repositories](/kb/semgrep-ci/bitbuket-jenkins-pipeline-projects/)
    
### Changed

> - The links within the [GitLab CI/CD config file](/semgrep-ci/sample-ci-configs/#sample-gitlab-cicd-configuration-snippet) have been updated.

Author: ```suggestion
- Updated the links within the [GitLab CI/CD config file](/semgrep-ci/sample-ci-configs/#sample-gitlab-cicd-configuration-snippet).
```

- Removed phone support from the docs.
- Updated the [Semgrep-Slack integration docs](/semgrep-cloud-platform/slack-notifications/) to clarify requirements for posting to private channels.
- Updated the [sample GHA config file](/writing-rules/private-rules/)for a CI job that publishes private Semgrep rules. 
- Clarified the Semgrep Assistant [privacy policy](/semgrep-code/semgrep-assistant-code/) on what data is stored.
- Updated [Semgrep Pro versus OSS](/docs/semgrep-pro-vs-oss/) docs. <!-- 1338 -->

### Fixed

> - Formatting on GitHub PR comments documentation. Thank you to [parsiya](https://github.com/parsiya) for the fix.

Author: ```suggestion
- Fixed formatting on GitHub PR comments documentation. Thank you to [parsiya](https://github.com/parsiya) for the fix.
```
